### DISCUSSION TITLE
Using IM to encapsulate PNG into SVG produces a problem

### DISCUSSION THREAD

**Discussion by schmidjd** [ImageMagick/ImageMagick#5218] [Development]
I recently used IM 7.0.8 to encapsulate a PNG into an SVG format (why do such a silly thing... long story).

I simply used `magick input.png output.svg` This produced an SVG file that displayed just fine in browsers (Edge, Chrome) and Affinity Designer.

But both Adobe Illustrator and Inkscape wouldn't display the image (Ai reported that there was a link to a non-existent file). But there was not supposed to be any link to an outside file... the PNG was encapsulated within the SVG,the line beginning with

`href="data:image/png;base64,iVBORw0KGgoA....`

To make it work, I had to edit that line to

`xlink:href="data:image/png;base64,iVBORw0KGgoA....`

So, why the prior "bad href" works fine in some tools and not in others is a bit baffling... but everyone's happy with the "good xlink:href" format.

Should this be fixed? Thanks... Jon

**Comment by snibgo** [comment]
I confirm the behaviour with IM v7.0.1-32 and Inkscape 1.0 (4035a4fb49, 2020-05-01).

However, the SVG documentation https://www.w3.org/TR/SVG/linking.html#XLinkHrefAttribute says:
> For backwards compatibility, elements with an ‘href’ attribute also recognize an ‘href’ attribute in the XLink namespace. [...] A conforming SVG generator must generate ‘href’ attributes without a namespace. However, it may also generate ‘href’ attributes in the XLink namespace to provide backwards compatibility.

I assume that IM is (or aspires to be) a conforming SVG generator. The bug seems to be in Inkscape and AI, not IM. Introducing a non-conforming feature (ie a bug) into IM seems like a bad idea. Perhaps IM should follow the "also" route, ie generate href without a namespace, and repeat it with a "xlink:" namespace, if that doesn't create other problems.

**Comment by urban-warrior** [comment]
Our preference is to adhere to standards, if we can, and wait for others to catch up.  Supporting both href and xlink would double the SVG image size which likely is undesirable behavior, particularly if the image is large (2X the disk space).
